utils/install.py

```diff
@@ -0,0 +1,1667 @@
+
+        # Check if wasmedge binary works
+        wasmedge_output = run_shell_command(
+            "{0}/bin/wasmedge --version".format(args.path)
+        )
+
+        if args.version in wasmedge_output:
+            print("WasmEdge Successfully installed")
+        else:
+            logging.critical(
+                "WasmEdge installation incorrect: {0}".format(wasmedge_output)
+            )
+
+        if IMAGE in args.extensions or "all" in args.extensions:
+            if install_image_extension(args, compat) != 0:
+                logging.error("Error in installing image extensions")
+            else:
+                print("Image extension installed")
+
+        if TENSORFLOW in args.extensions or "all" in args.extensions:
+            if install_tensorflow_extension(args, compat) != 0:
+                logging.error("Error in installing tensorflow extensions")
+            else:
+                print("Tensorflow extension installed")
+
+        install_plugins(args, compat)
+
+        # Cleanup
+        shutil.rmtree(TEMP_PATH)
+
+        print("Run:\nsource {0}".format(CONST_shell_config))
+    else:
+        reraise(Exception("Incompatible with your machine\n{0}".format(compat)))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="WasmEdge installation, uninstallation and extensions install"
+    )
+    parser.add_argument(
+        "-e",
+        "--extension",
+        dest="extensions",
+        choices=EXTENSIONS.append("all"),
+        required=False,
+        default=[],
+        nargs="*",
+        help="Supported Extensions - {0}".format(EXTENSIONS),
+    )
+    parser.add_argument(
+        "-v",
+        "--version",
+        dest="version",
+        default=get_latest_github_release("WasmEdge/WasmEdge"),
+        required=False,
+        help="Version for WasmEdge",
+    )
+    parser.add_argument(
+        "-D",
+        "--debug",
+        dest="loglevel",
+        required=False,
+        action="store_const",
+        const=logging.DEBUG,
+        help="Verbosity debug",
+    )
+    parser.add_argument(
+        "-p",
+        "--path",
+        dest="path",
+        required=False,
+        default=PATH,
+        help="Installation path for WasmEdge",
+    )
+    parser.add_argument(
+        "-r",
+        "--remove-old",
+        dest="remove_old",
+        required=False,
+        choices=["yes", "no"],
+        help="Run uninstaller script before installing",
+    )
+    parser.add_argument(
+        "-u",
+        "--uninstall-script-tag",
+        dest="uninstall_script_tag",
+        required=False,
+        default=get_latest_github_release("WasmEdge/WasmEdge"),
+        help="GitHub tag for uninstall script",
+    )
+    parser.add_argument(
+        "--plugins",
+        dest="plugins",
+        required=False,
+        default=[],
+        nargs="*",
+        help="Supported Plugins. Example\n"
```

Comment:
Please add an experimental option warning, we may change this behavior once the plugin metadata is finished.

Reply:
Alright